File: src/main/java/com/svantle/score/listeners/PlayerLeaveListener.java
package com.svantle.score.listeners;

import com.svantle.score.SCore;
import org.bukkit.Sound;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

public class PlayerLeaveListener implements Listener {

    private final SCore plugin;

    public PlayerLeaveListener(SCore plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onPlayerLeave(PlayerQuitEvent event) {
        Player player = event.getPlayer();


        event.setQuitMessage(null);

        Sound leaveSound = Sound.valueOf(plugin.getConfigHandler().getConfig().getString("leave-sound", ""));
        player.playSound(player.getLocation(), leaveSound, 1.0F, 1.0F);

        player.sendMessage(plugin.getMessagesHandler().get("player-leave"));
    }
}

File: src/main/kotlin/com/svantle/score/listeners/PlayerLeaveListener.kt
package com.svantle.score.listeners

import com.svantle.score.SCore
import com.svantle.score.constants.CONFIG_JOIN_SOUND
import com.svantle.score.constants.CONFIG_LEAVE_SOUND
import com.svantle.score.constants.MSG_PLAYER_JOIN
import com.svantle.score.constants.MSG_PLAYER_LEAVE
import com.svantle.score.utils.MessageHelper
import com.svantle.score.utils.playTo
import com.svantle.score.utils.sendTo
import org.bukkit.Sound
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent

class PlayerLeaveListener(private val core: SCore): Listener {

	init {
		core.server.pluginManager.registerEvents(this, core)
	}

	@EventHandler
	fun onQuit(event: PlayerQuitEvent) {
		event.quitMessage(null)

		Sound.valueOf(core.config.get(CONFIG_LEAVE_SOUND)).playTo(event.player)
		MessageHelper.phrase(MSG_PLAYER_LEAVE).sendTo(event.player)
	}
}